src/lightning_app/testing/testing.py

```diff
@@ -366,68 +379,51 @@ def run_app_in_cloud(
         admin_page.locator(f'[data-cy="{name}"]').click()
 
         client = LightningClient()
-        project = _get_project(client)
-
-        lit_apps = [
-            app
-            for app in client.lightningapp_instance_service_list_lightningapp_instances(
-                project_id=project.project_id
-            ).lightningapps
-            if app.name == name
-        ]
-        if not lit_apps:
-            return True
-        assert len(lit_apps) == 1
-        app = lit_apps[0]
+        project_id = _get_project(client).project_id
+
+        app = _fetch_app_by_name(client, project_id, name)
         app_id = app.id
         print(f"The Lightning App ID is: {app.id}")  # useful for Grafana
 
         if debug:
             process = Process(target=_print_logs, kwargs={"app_id": app_id})
             process.start()
 
-        # Wait until the app is running
+        i = 1
         while True:
-            sleep(1)
-
-            lit_apps = [
-                app
-                for app in client.lightningapp_instance_service_list_lightningapp_instances(
-                    project_id=project.project_id
-                ).lightningapps
-                if app.name == name
-            ]
-            app = lit_apps[0]
-
+            app = _fetch_app_by_name(client, project_id, name)
+            # wait until the app is running
             if app.status.phase == V1LightningappInstanceState.RUNNING:
+                print("App is running, continuing with testing...")
+                view_page = context.new_page()
+                app_url = app.status.url
+                view_page.goto(f"{app_url}/view")
+                j = 1
+                status_code = None
+                while status_code != 200:
+                    status_code = requests.get(app_url + "/openapi.json").status_code
+                    if debug and j % 30 == 0:
+                        print(f"Received status code {status_code} at {app_url!r}, continuing infinite loop...")
+                    j += 1
+                    sleep(1)
                 break
-
-        view_page = context.new_page()
-        view_page.goto(f"{app.status.url}/view")
-
-        # TODO: is re-creating this redundant?
-        lit_apps = [
-            app
-            for app in client.lightningapp_instance_service_list_lightningapp_instances(
-                project_id=project.project_id
-            ).lightningapps
-            if app.name == name
-        ]
-        app = lit_apps[0]
-        app_url = app.status.url
-        while True:
-            sleep(1)
-            resp = requests.get(app_url + "/openapi.json")
-            if resp.status_code == 200:
+            if app.status.phase == V1LightningappInstanceState.STOPPED:
```

Comment:
Maybe this should be:
```suggestion
            elif app.status.phase != V1LightningappInstanceState.PENDING:
```
If it isn't pending then it started at some point then either stopped or failed - either way we can break out of the loop